app/src/main/java/com/sunasterisk/musixmatch/data/source/PlaylistDataSource.java

```diff
@@ -17,13 +17,34 @@
 
     Uri addSongToPlaylist(long Id, Track track);
 
-    long createPlaylist(String name);
+    void createPlaylist(String name, onCreatePlaylist callback);
 
     long getPlaylistId(String name);
 
-    int deletePlaylist(long id);
+    void deletePlaylist(long id, onDeletePlaylist callback);
 
     int removeTrackFromPlaylist(long id, Track track);
 
-    int renamePlaylist(long id, String newName);
+    void renamePlaylist(long id, String newName, onRenamePlaylist callback);
+
+    interface onCreatePlaylist {
```

Comment:
Tên class không nên để chữ cái thường nên để chữ in hoa và là danh từ. Nếu là interface thì thì nên thêm Listener, Callback... hậu tố vào sau cho nó có ý nghĩa.

Reply:
interface này của em k có handle gì về click ạ, chỉ nhận notify
vậy em có thể đặt tên sao cho có ý nghĩa ạ!
em cảm ơn.

Reply:
suggest thôi nhé có thể đặt là CreatingPlayListCallback